code/script.js

```diff
@@ -220,106 +226,154 @@ const generateBoard = () => {
+    // Defines the condition for 'accessories' or 'other' category.  
   } else if (category === 'accessories' || category === 'other') {
-
+      keep = secret[category].includes(value);
   }
-}
+  // This invoke filterCharacters
+  filterCharacters(keep)
+};
 
-// It'll filter the characters array and redraw the game board.
+// This will filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const {category, value} = currentQuestion
+  
   // Show the correct alert message for different categories
   if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+        `Yes, the person wears ${value}! Keep all people that wears ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+        `No, the person doesn't wear ${value}! Remove all people that wears ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'other') { 
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person is a ${value}! Keep all people that is a ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person isn't a ${value}! Remove all people that isn't a ${value}.`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  } else if (category === 'hair') {
+      if (keep) {
+      alert(
+        `Yes, the person has ${value} hair! Keep all the people with ${value} hair.`
+        );
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
```

Comment:
Maybe you could have added an explanatory comment on how you're using the filter method to remove or keep characters?